protocol/notifications.proto

```diff
@@ -0,0 +1,334 @@
+  rpc SetSpaceSettings(SetSpaceSettingsRequest) returns (SetSpaceSettingsResponse);
+  // SetSpaceChannelSettings updates settings for a channel that is part of a space.
+  rpc SetSpaceChannelSettings(SetSpaceChannelSettingsRequest) returns (SetSpaceChannelSettingsResponse);
+  // SubscribeWebPush subscribes a client to receive notifications through web push.
+  // If the given req.subscription is already associated with a web push subscription the user id is updated (upsert).
+  rpc SubscribeWebPush(SubscribeWebPushRequest) returns (SubscribeWebPushResponse);
+  // UnsubscribeWebPush deletes web push subscription.
+  rpc UnsubscribeWebPush(UnsubscribeWebPushRequest) returns (UnsubscribeWebPushResponse);
+  // SubscribeAPN subscribes a device to receive Apple Push Notifications.
+  // If the given device token is already associated with an APN subscription the user id is updated (upsert).
+  rpc SubscribeAPN(SubscribeAPNRequest) returns (SubscribeAPNResponse);
+  // UnsubscribeAPN unsubscribes a device from receiving Apple Push Notifications.
+  rpc UnsubscribeAPN(UnsubscribeAPNRequest) returns (UnsubscribeAPNResponse);
+}
+
+// DmChannelSettingValue specifies if the user wants to receive notifications for DM streams.
+// This can be overwritten with DM channel specific configuration.
+enum DmChannelSettingValue {
+  // DM_UNSPECIFIED not set, assumes DM_MESSAGES_YES as a default.
+  DM_UNSPECIFIED = 0;
+  // DM_MESSAGES_YES indicates that the user wants to receive notifications for DM channels.
+  DM_MESSAGES_YES = 1;
+  // DM_MESSAGES_NO indicates that the user doesn't want to receive notifications for DM channels.
+  DM_MESSAGES_NO = 2;
+  // DM_MESSAGES_NO_AND_MUTE indicates that the user doesn't want to receive notifications for DM channels
+  // and the UI should not render any feedback that a message was received.
+  DM_MESSAGES_NO_AND_MUTE = 3;
+}
+
+// GdmChannelSettingValue is a default setting that is applied to all GDM streams a user is in and can be
+// overwritten by GDM channel specific settings.
+enum GdmChannelSettingValue {
+  // GDM_UNSPECIFIED not set, assumes GDM_MESSAGES_ALL as a default.
+  GDM_UNSPECIFIED = 0;
+  // GDM_NO_MESSAGES indicates that the user will not receive notifications for GDM channels.
+  GDM_NO_MESSAGES = 1;
+  // GDM_NO_MESSAGES_AND_MUTE indicates that the user doesn't want to receive notifications for GDM channels
+  // and the UI should not render any feedback that a message was received.
+  GDM_NO_MESSAGES_AND_MUTE = 2;
+  // GDM_ONLY_MENTIONS_REPLIES_REACTIONS indicates that the user receives notifications for messages added to
+  // GDM channels that are either a direct reply or a reaction to his own messages.
+  GDM_ONLY_MENTIONS_REPLIES_REACTIONS = 3;
+  // GDM_MESSAGES_ALL indicates that the user receives notifications for all messages in GDM channels.
+  // This is the default.
+  GDM_MESSAGES_ALL = 4;
+}
+
```

Comment:
Can we fix these enums to be consistent? We've got `GDM_NO_MESSAGES` with no first, and `DM_MESSAGES_NO` with no last...

I wasn't going to bring this up, but do we really need a new enum for each type? I think the ui can just not show the mentions option in dms... seems like `unspecified, all, mentions_replies_reactions, none, muted` would work... if not it's fine.

Reply:
> Can we fix these enums to be consistent? We've got `GDM_NO_MESSAGES` with no first, and `DM_MESSAGES_NO` with no last...
> 
> I wasn't going to bring this up, but do we really need a new enum for each type? I think the ui can just not show the mentions option in dms... seems like `unspecified, all, mentions_replies_reactions, none, muted` would work... if not it's fine.

The definitions are now renamed for consistency. Personally I prefer to have separate enums for them.

Reply:
I trust you, but can you tell me more about your thought process so that I can be on the same page. Also, heads up - we will probably add a new notification type for enabling disabling notifications about tipping in the very near future.

Reply:
My reasoning is that having a single enum for a single purpose is easier to understand instead of sharing them and filtering out the ones that aren't supported in the context.

Reply:
I don't think you need to filter on the server. I think you can treat every stream the same. On the client we have made the UX decision that replies and mentions don't make sense in DMs. but that is purely a UX decision and frankly, might change one day in the future.